transformer_engine/common/transpose/cast_transpose_fusion.cu

```diff
@@ -568,16 +464,46 @@ cast_transpose_fused_kernel_notaligned(const Param param,
     }
 
     /* warp tile amax reduce*/
-    max = reduce_max<cast_transpose_num_threads / THREADS_PER_WARP>(max, warp_id);
+    amax = reduce_max<cast_transpose_num_threads / THREADS_PER_WARP>(amax, warp_id);
 
     if (threadIdx.x == 0) {
         static_assert(std::is_same<CType, float>::value);
         if (param.amax != nullptr) {
-            atomicMaxFloat(param.amax, max);
+            atomicMaxFloat(param.amax, amax);
         }
     }
 }
 
+static const char* ActTypeToString[] = {
+    "NoAct",    // 0
+    "Sigmoid",  // 1
+    "GeLU",     // 2
+    "QGeLU",    // 3
+    "SiLU",     // 4
+    "ReLU",     // 5
+    "SReLU"     // 6
+};
+
+template <typename ComputeType, typename ParamOP,
+          ComputeType (*OP)(ComputeType, const ParamOP&)>
+int get_dactivation_type() {
+    if (OP == &sigmoid<ComputeType, ComputeType>) {
+        return 1;
+    } else if (OP == &dgelu<ComputeType, ComputeType>) {
+        return 2;
+    } else if (OP == &dqgelu<ComputeType, ComputeType>) {
+        return 3;
+    } else if (OP == &dsilu<ComputeType, ComputeType>) {
+        return 4;
+    } else if (OP == &drelu<ComputeType, ComputeType>) {
+        return 5;
+    } else if (OP == &dsrelu<ComputeType, ComputeType>) {
+        return 6;
+    } else {
+        return 0;
+    }
+}
```

Comment:
Hi. You may want to have a look at the `NVTE_Activation_Type` in `common/include/transformer_engine/activation.h` and see if you can use that enum here.

Reply:
Hi Phuong, thanks a lot for the review and the comment. I completely agree that enum would be the way to go, but its usage becomes a problem during JIT compilation. The RTC manager converts kernel input parameters into strings and then concatenates them into a single kernel label. This process is more complicated and convoluted on the both sides, i.e., from the calling routine and from the to-be-compiled-code represented as a string. Passing such a enum type became problematic, so I decided to follow the most convenient way with using the mapping from an activation function pointer to an integer number